x/wasm/internal/keeper/keeper.go

```diff
@@ -189,7 +218,12 @@ func (k Keeper) Instantiate(ctx sdk.Context, codeID uint64, creator, admin sdk.A
 
 	// persist instance
 	createdAt := types.NewCreatedAt(ctx)
-	instance := types.NewContractInfo(codeID, creator, admin, initMsg, label, createdAt)
+
+	modifyConfig := types.AllowNobody // TODO: How about default to creator?
```

Comment:
Default should be immutable. We only make "upgradeable contracts" when explicitly requested. This is an extra feature

Reply:
That said, with my previous comments, we can likely store the admin on contractInfo as now, and just update the store and instantiate permisisons as you already did